lib/checks.js

```diff
@@ -152,5 +152,14 @@ var commands = module.exports = {
 
       return result;
     }
+  },
+
+  is_valid_shell: function(args) {
+    // Dockerfile syntax requires that args be in JSON array format
+    var parsed = JSON.parse(args);
+    if (parsed.constructor !== Array) {
```

Comment:
Is it worth checking that all entries in array are strings?  Can't test this command with my version of docker.